src/journal-remote/journal-upload.c

```diff
@@ -226,6 +230,23 @@ int start_upload(Uploader *u,
                         h = l;
                 }
 
+                char **values;
+                const char *name;
+                ORDERED_HASHMAP_FOREACH_KEY(values, name, arg_headers) {
+                        _cleanup_free_ char *joined = strv_join(values, ", ");
+                        if (!joined)
+                                return log_oom();
+
+                        _cleanup_free_ char *header = strjoin(name, ": ", joined);
```

Comment:
An empty value string is allowed? If allowed, and `joined` is an empty string, then still `": "` is necessary?

Reply:
colon is still necessary

Reply:
How about the trailing space in that case?

Reply:
https://www.rfc-editor.org/rfc/rfc2616#section-4.2
```
The field value MAY be preceded by any amount of LWS, though a single SP is preferred
```

Reply:
Please also check the validity of the concatenated string with `header_value_is_valid()`.

Reply:
concatenated header contains pair and header_value_is_valid checks validates value only

Reply:
I mean, `header_value_is_valid(joined)`.